frame/clearing-house/src/lib.rs

```diff
@@ -465,44 +478,37 @@ pub mod pallet {
 			Ok(())
 		}
 
-		fn create_market(
-			asset: Self::AssetId,
-			vamm_params: Self::VammParams,
-			margin_ratio_initial: Self::Decimal,
-			margin_ratio_maintenance: Self::Decimal,
-			funding_frequency: Self::DurationSeconds,
-			funding_period: Self::DurationSeconds,
-		) -> Result<Self::MarketId, DispatchError> {
-			MarketCount::<T>::try_mutate(|id| {
-				ensure!(T::Oracle::is_supported(asset)?, Error::<T>::NoPriceFeedForAsset);
-				ensure!(
-					funding_period > 0 && funding_frequency > 0,
-					Error::<T>::ZeroLengthFundingPeriodOrFrequency
-				);
-				ensure!(
-					funding_period.rem_euclid(funding_frequency) == 0,
-					Error::<T>::FundingPeriodNotMultipleOfFrequency
-				);
-				ensure!(
-					margin_ratio_initial > T::Decimal::zero() &&
-						margin_ratio_initial < T::Decimal::one() &&
-						margin_ratio_maintenance > T::Decimal::zero() &&
-						margin_ratio_maintenance < T::Decimal::one(),
-					Error::<T>::InvalidMarginRatioRequirement
-				);
-				ensure!(
-					margin_ratio_initial > margin_ratio_maintenance,
-					Error::<T>::InitialMarginRatioLessThanMaintenance
-				);
+		fn create_market(config: &Self::MarketConfig) -> Result<Self::MarketId, DispatchError> {
+			ensure!(T::Oracle::is_supported(config.asset)?, Error::<T>::NoPriceFeedForAsset);
+			ensure!(
+				config.funding_period > 0 && config.funding_frequency > 0,
+				Error::<T>::ZeroLengthFundingPeriodOrFrequency
+			);
+			ensure!(
+				config.funding_period.rem_euclid(config.funding_frequency) == 0,
+				Error::<T>::FundingPeriodNotMultipleOfFrequency
+			);
+			ensure!(
+				config.margin_ratio_initial > T::Decimal::zero() &&
+					config.margin_ratio_initial < T::Decimal::one() &&
+					config.margin_ratio_maintenance > T::Decimal::zero() &&
+					config.margin_ratio_maintenance < T::Decimal::one(),
+				Error::<T>::InvalidMarginRatioRequirement
+			);
+			ensure!(
+				config.margin_ratio_initial > config.margin_ratio_maintenance,
+				Error::<T>::InitialMarginRatioLessThanMaintenance
+			);
```

Comment:
One thing that I want to go over in a future office hours is the use of validation objects to encapsulate all validity checks. I'll leave some important links to read more about them; maybe I'll discuss it more in depth on Tuesday. [[composable-support::src::validation.rs](https://github.com/ComposableFi/composable/blob/main/frame/composable-support/src/validation.rs); [vault::src::validation.rs](https://github.com/ComposableFi/composable/blob/main/frame/vault/src/validation.rs)]

Reply:
this hasn't been applied to many pallets, so there aren't a lot of examples rn

Reply:
Cool. Seems similar to the `Context`s used in Solana's Anchor framework